tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -5,41 +5,46 @@
 using System.Globalization;
 using System.Threading;
 using System.Collections.Generic;
+using Xamarin.Localization.MSBuild;
 
 namespace Xamarin.iOS.Tasks {
 	[TestFixture]
 	public class LocalizationStringTest : TestBase {
-		[TestCase ("cs-CZ", "došlo k chybě: neznámý formát image")]
-		[TestCase ("de-DE", "Unbekanntes Imageformat.")]
-		[TestCase ("en-US", "Unknown image format.")]
-		[TestCase ("es-ES", "formato de imagen desconocido.")]
-		[TestCase ("fr-FR", "format d'image inconnu.")]
-		[TestCase ("it-IT", "Formato immagine sconosciuto.")]
-		[TestCase ("ja-JP", "の読み込みでエラーが発生しました: 画像の形式が不明です。")]
-		[TestCase ("ko-KR", "을(를) 로드하는 동안 오류 발생: 알 수 없는 이미지 형식입니다.")]
-		[TestCase ("pl-PL", "nieznany format obrazu.")]
-		[TestCase ("pt-BR", "formato de imagem desconhecido.")]
-		[TestCase ("ru-RU", "неизвестный формат изображения.")]
-		[TestCase ("tr-TR", "yüklenirken hata oluştu: Görüntü biçimi bilinmiyor.")]
-		[TestCase ("zh-CN", "时出错: 未知图像格式")]
-		[TestCase ("zh-TW", "時發生錯誤: 未知的映像格式。")]
-		public void AllSupportedTranslations (string culture, string errorMessage)
+		[TestCase ("cs-CZ")]
+		[TestCase ("de-DE")]
+		[TestCase ("es-ES")]
+		[TestCase ("fr-FR")]
+		[TestCase ("it-IT")]
+		[TestCase ("ja-JP")]
+		[TestCase ("ko-KR")]
+		[TestCase ("pl-PL")]
+		[TestCase ("pt-BR")]
+		[TestCase ("ru-RU")]
+		[TestCase ("tr-TR")]
+		[TestCase ("zh-CN")]
+		[TestCase ("zh-TW")]
+		public void SpecificErrorTranslation (string culture)
 		{
+			// insert which error code you'd like to test
+			string errorCode = "E0007";
 			CultureInfo originalCulture = Thread.CurrentThread.CurrentUICulture;
-			CultureInfo newCulture;
+
 			try {
-				newCulture = new CultureInfo (culture);
-				Thread.CurrentThread.CurrentUICulture = newCulture;
-				var task = CreateTask<CollectITunesArtwork> ();
-				task.ITunesArtwork = new TaskItem [] { new TaskItem (Assembly.GetExecutingAssembly ().Location) };
+				string englishError = TranslateError("en-US", errorCode);
+				string newCultureError = TranslateError(culture, errorCode);
 
-				Assert.IsFalse (task.Execute (), "Execute failure");
-				Assert.AreEqual (1, Engine.Logger.ErrorEvents.Count, "ErrorCount");
-				bool isTranslated = Engine.Logger.ErrorEvents[0].Message.Contains (errorMessage);
-				Assert.IsTrue (isTranslated, culture + ": is not supported correctly. ");
+				Assert.AreNotEqual (englishError, newCultureError, $"\"{errorCode}\" is not translated in {culture}.");
```

Comment:
this will not fail if the translation string is null or empty
is there something (in the code) that ensure the English (default) is returned if no translation is available ?

Reply:
Very embarrassed that I didn't check for null or empty strings but that and handling non-existent error codes should be taken care of now!